### PR TITLE
vimc-4897: Add vignette to compare impact functions

### PR DESCRIPTION
**Pull request by r-ash**
Note this is merging into #19 

I'm expecting this vignette will be removed once we only have 1 method for calculating impact

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ docs
 .Renviron
 .Rhistory
 .DS_Store
+vignettes_src/internal-impact.md
```
Comment (richfitz) [inline .gitignore]: You might prefer

```
vignettes_src/*.html
vignettes_src/*.md
```

File: vignettes/internal-impact.Rmd [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+or use `method = "yov_birth_cohort` for impact by year of vaccination stratified by birth cohort. This uses the same logic as other impact methods but before calling the public facing impact function it will also extract FVP data from the data base for this `touchstone` and vaccine delivery. It then calls either `impact_by_year_of_vaccination_activity_type` or `impact_by_year_of_vaccination_birth_cohort` to get impact.
+
+### Recipe interface
+
+We can re-use the recipe from above and get meta table for this method
+
+
+```r
+meta <- vimpact:::get_meta_from_recipe(default_recipe = FALSE, recipe = t,
```
Comment (richfitz) [inline vignettes/internal-impact.Rmd:120]: should we be advocating using private access here? `:::` used in a few places

**Review by richfitz — CHANGES_REQUESTED** [review]
Main concern is the use of `:::` here; seems not ideal unless these are waiting to be exported later (or are about to be removed?)

File: vignettes/internal-impact.Rmd (hunk 2 above)
Comment (r-ash) [reply vignettes/internal-impact.Rmd:120]: Probably not but this is one of Xiang's functions which will be removed eventually

**Review by richfitz — APPROVED** [review]

**Pull request merged**